game/core/game_view.py

```diff
@@ -204,7 +221,16 @@ def reload_modules(self) -> None:
                 self.registered_sprites.append(register.sprite)
 
         self._reload_module(self.raft_module, self.vehicle_register)
-        self._reload_module(self.player_module, self.player_register)
+
+        # HACK: This is for checking if Task 1 is complete
+        filename1 = self._reload_player_and_get_sheet_name()
+        filename2 = self._reload_player_and_get_sheet_name()
+        filename3 = self._reload_player_and_get_sheet_name()
+        if filename1 == filename2 and filename2 == filename3:
+            self.disable_movement = False
+        else:
+            self.disable_movement = True
```

Comment:
This is pretty clever!

You could have it run just once and consider some minor renaming to make this more clear:

```suggestion
        # Prevent motion until a single character asset is specified
        if self.disable_movement:
            asset1 = self._reload_player_and_get_sheet_name()
            asset2 = self._reload_player_and_get_sheet_name()
            asset3 = self._reload_player_and_get_sheet_name()
            self.disable_movement = not (asset1 == asset2 == asset3)
```

Reply:
Yeah I intentionally didn't want to run it just once in case the player reverts their change, but I do feel like there's no need to run it every time 👍